Mathlib/LinearAlgebra/TensorProduct/Subalgebra.lean

```diff
@@ -128,23 +126,101 @@ theorem comm_trans_rTensorBot :
     (Algebra.TensorProduct.comm R _ _).trans A.rTensorBot = A.lTensorBot :=
   AlgEquiv.toLinearEquiv_injective (toSubmodule A).comm_trans_rTensorOne
 
+end Subalgebra
+
+namespace Algebra.TensorProduct
+
+variable (R S T)
+
+/-- Given `R`-algebras `S,T`, there is a natural `R`-linear isomorphism from `S ⊗[R] T` to
+`S' ⊗[R] T'` where `S',T'` are images of `S,T` in `S ⊗[R] T`.
+This is promoted to an `R`-algebra isomorphism `Algebra.TensorProduct.algEquivIncludeRange`. -/
+def linearEquivIncludeRange :
+    S ⊗[R] T ≃ₗ[R] (includeLeft : S →ₐ[R] S ⊗[R] T).range ⊗[R]
+      (includeRight : T →ₐ[R] S ⊗[R] T).range := .ofLinear
+  (_root_.TensorProduct.map
+    includeLeft.toLinearMap.rangeRestrict includeRight.toLinearMap.rangeRestrict)
+  (Submodule.mulMap (LinearMap.range includeLeft) (LinearMap.range includeRight))
```

Comment:
You may or may not want to use `.mulMap` dot notation here (and another place below).

Reply:
I think it's OK to use dot notation here.
```suggestion
  ((LinearMap.range includeLeft).mulMap (LinearMap.range includeRight))
```